### ISSUE TITLE
Geolocation can not watch the position.

### ISSUE THREAD

**Issue by keung725** [facebook/react-native#9347]
Geolocation can not watch the position. It only updates when you go to homescreen and reopen the app. To recreate this problem, just use the example on React Native docs: 
http://facebook.github.io/react-native/releases/0.31/docs/geolocation.html#watchposition

```
componentDidMount(){
  this.watchID = navigator.geolocation.watchPosition((position) => {
      var lastPosition = JSON.stringify(position);
      this.setState({lastPosition});
    });
}
```

Using React Native version 0.31.0 & 0.30.0 on iOS.

**Comment by ericvicenti** [comment]
I just tested with the latest RN on the iOS simulator, and it works fine in this case. (I haven't tried on a real device)

Are you still having this issue?

**Comment by ericvicenti** [comment]
Closing for now. Please re-open if this problem is still here!

**Comment by jaredtibs** [comment]
@ericvicenti 

I am experiencing this issue with:
```
react 15.4.2
react-native 0.41.1
xcode 8.2.1
os sierra
ios 10.2
```

It only loads the first location on boot, but then does not register any change in location when using Debug -> Location. Any help at all would be greatly appreciated.

**Comment by GurtejSafaltek** [comment]
yes I'm facing same issue @jaredtibs

**Comment by edencakir** [comment]
I'm having the same issue as well. @ericvicenti

It works fine on emulator, stops updating on real device. Do you guys have a guess about it?

**Comment by nos-joel** [comment]
The same issue is happening to me. It only refreshes when you refresh the app. The location does not refresh real time. See this expo snack and test with a real device. https://snack.expo.io/SJb5ZGMd-

**Comment by ckim16** [comment]
I'm using a real android device and `watchPosition` is not event getting invoked.

```
"react": "16.0.0-alpha.12",
"react-native": "0.45.1",
```

**Comment by leonardodino** [comment]
same here, should we open a new issue? seems like a regression? @ericvicenti

**Comment by ckim16** [comment]
Little update here. `watchPosition` inside of `componentDidMount` does get invoked.

After about 5~10 mins after the component gets mounted.

**Comment by ckim16** [comment]
Remove `enableHighAccuracy: true` option and try it. It works for me.

**Comment by edencakir** [comment]
@ckim16 what if we need that option enabled?

**Comment by leonardodino** [comment]
testing in the ios simulator even with this disabled doesn't works... 😫

**Comment by ckim16** [comment]
@edencakir I'm sorry I really do not know what to do. I just wanted to make that `watchPosition` part work in my android device and it worked after I removed that `enableHighAccuracy` option. For now it will stay in that way and hopefully someone will insight us with the better answer.

@leonardodino That's really weird and I'm sorry to hear that. My ios, both simulator and device, has no trouble using watchPosition. Hopefully you have found your answer by now!

**Comment by jason2li** [comment]
Any update on this?  I'm having the same issues.

**Comment by RWOverdijk** [comment]
Same here. It only calls when the app opens. So suspending it (going to home) and opening the app again also triggers the listeners. But it doesn't update at all otherwise.

**Comment by briangainer** [comment]
I have the same issue, watchPosition() is not updating. In my project, it gets called 2 or 3 times after my screen is mounted and then stops.

**Comment by bakerjay06074** [comment]
Having the same problem with watchposition running the snack posted above by @nos-joel.  But if I run watchposition in a simple PhoneGap app, it works fine, with updates once per second while I'm driving.  Really frustrating.  I wanted to use React Native for the UI benefits, but I can't make it work.

**Comment by gitsad** [comment]
I have the same problem which @briangainer has. @ericvicenti I think this issue should be open.

**Comment by pulkitsapra** [comment]
Someone please provide a solution. Watchposition invokes only once inside ComponentDidMount () when homepage/screen starts
